rpg.py

```diff
@@ -143,7 +139,10 @@ def use_item(self):
             item_input = input('>')
             if item_input == 'y':
                 if self.hp < self.max_hp:
-                    self.hp += potion.hp_restore
+                    for i in range(0, potion.hp_restore + 1):
```

Comment:
You want to use `min()`
Using a `for` loop to do this is very bad in terms of runtime complexity (scales with `self.max_hp`).
When the numbers get big, this method becomes insanely slow.